core/util.go

```diff
@@ -478,6 +478,36 @@ func VerifyCSR(csr *x509.CertificateRequest) error {
 	return errors.New("Unsupported CSR signing algorithm")
 }
 
+var (
+	oidExtensionRequest = asn1.ObjectIdentifier{1, 2, 840, 113549, 1, 9, 14}
+	oidSubjectAltName   = asn1.ObjectIdentifier{2, 5, 29, 17}
+)
+
+// ExtensionsFromCSR extracts the set of requested extensions from a CSR
+func ExtensionsFromCSR(csr *x509.CertificateRequest) map[string][]byte {
+	extensionMap := map[string][]byte{}
+	for _, attr := range csr.Attributes {
```

Comment:
Why are we using Attributes here? The docs say:

>    // Attributes is a collection of attributes providing
>    // additional information about the subject of the certificate.
>    // See RFC 2986 section 4.1.

I think we want the Extensions field.